src/Microsoft.AspNetCore.Mvc.Razor.Extensions/DeclarationOnlyExtensionInitializer.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Extensions
+{
+    internal class DeclarationOnlyExtensionInitializer : ExtensionInitializer
```

Comment:
Are you doing this for testing? Nothing MVC does needs 2-phase build.

Reply:
Ah - I see, you're doing this in preparation for mainlining the 3.0 compiler.

Reply:
Components in 3.0 need to go in the mainline compiler - not as an MVC feature. I think what you're doing here is OK but  Components and their codegen will just be part of the compiler.